status-im/status-react
# Activate queue message behavior for more intuitive message order

Fixes #2702 

See https://github.com/status-im/status-react/pull/2743 for ground work.

### Summary

Intercept received messages and queue them

- Exposes behavior under QUEUE_MESSAGE_ENABLED flag that is turned off in
production.

- Delays messages to allow for reordering of latecoming messages up to 500ms

### Review notes (optional):
Normal review.

### Testing notes (optional):
1. Make a build with QUEUE_MESSAGE_ENABLED enabled and one without
2. Send lots of messages in 1on1 chat from one person, like `1 2 3 4 5 6 7 8 9`.
3a. With flag off you should eventually see ordering being different due to network uncertainty
3b. With flag on this should essentially be impossible to achieve (or way less likely)

status: ready
